In [13]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.model_selection import train_test_split, StratifiedKFold, cross_val_score

from sklearn.pipeline import Pipeline
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import (
    accuracy_score, precision_score, recall_score, f1_score,
    roc_auc_score, confusion_matrix, classification_report
)


# 1. EDA

## 1.1 Upload the dataset

In [34]:
data=pd.read_csv("dataR2.csv")

In [35]:
data.head()

Age        BMI  Glucose  Insulin      HOMA   Leptin  Adiponectin  Resistin  \
0   48  23.500000       70    2.707  0.467409   8.8071     9.702400   7.99585   
1   83  20.690495       92    3.115  0.706897   8.8438     5.429285   4.06405   
2   82  23.124670       91    4.498  1.009651  17.9393    22.432040   9.27715   
3   68  21.367521       77    3.226  0.612725   9.8827     7.169560  12.76600   
4   86  21.111111       92    3.549  0.805386   6.6994     4.819240  10.57635   

     MCP.1  Classification  
0  417.114               1  
1  468.786               1  
2  554.697               1  
3  928.220               1  
4  773.920               1

There are 9 features: age, BMI, glucose, insulin, HOMA, leptin, adiponectin, resistin, MCP.1, all quantitative. 
The target variable returns 1 if the patient is healthy, 2 if the patient is diagnosed with breast cancer. 

In [36]:
data["target"] = data["Classification"].map({1: 0, 2: 1})
data

Age        BMI  Glucose  Insulin      HOMA   Leptin  Adiponectin  \
0     48  23.500000       70    2.707  0.467409   8.8071     9.702400   
1     83  20.690495       92    3.115  0.706897   8.8438     5.429285   
2     82  23.124670       91    4.498  1.009651  17.9393    22.432040   
3     68  21.367521       77    3.226  0.612725   9.8827     7.169560   
4     86  21.111111       92    3.549  0.805386   6.6994     4.819240   
..   ...        ...      ...      ...       ...      ...          ...   
111   45  26.850000       92    3.330  0.755688  54.6800    12.100000   
112   62  26.840000      100    4.530  1.117400  12.4500    21.420000   
113   65  32.050000       97    5.730  1.370998  61.4800    22.540000   
114   72  25.590000       82    2.820  0.570392  24.9600    33.750000   
115   86  27.180000      138   19.910  6.777364  90.2800    14.110000   

     Resistin    MCP.1  Classification  target  
0     7.99585  417.114               1       0  
1     4.06405  468.786               1       0  
2     9.27715  554.697               1       0  
3    12.76600  928.220               1       0  
4    10.57635  773.920               1       0  
..        ...      ...             ...     ...  
111  10.96000  268.230               2       1  
112   7.32000  330.160               2       1  
113  10.33000  314.050               2       1  
114   3.27000  392.460               2       1  
115   4.35000   90.090               2       1  

[116 rows x 11 columns]

In [37]:
data.isnull().any()

Age               False
BMI               False
Glucose           False
Insulin           False
HOMA              False
Leptin            False
Adiponectin       False
Resistin          False
MCP.1             False
Classification    False
target            False
dtype: bool

In [38]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 116 entries, 0 to 115
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             116 non-null    int64  
 1   BMI             116 non-null    float64
 2   Glucose         116 non-null    int64  
 3   Insulin         116 non-null    float64
 4   HOMA            116 non-null    float64
 5   Leptin          116 non-null    float64
 6   Adiponectin     116 non-null    float64
 7   Resistin        116 non-null    float64
 8   MCP.1           116 non-null    float64
 9   Classification  116 non-null    int64  
 10  target          116 non-null    int64  
dtypes: float64(7), int64(4)
memory usage: 10.1 KB


In [39]:
data['target'].value_counts()

1    64
0    52
Name: target, dtype: int64

$$
\text{Class 0 (Healthy)} \rightarrow \frac{52}{116} \approx 44.8\% \\
\text{Class 1 (Breast cancer)} \rightarrow \frac{64}{116} \approx 55.2\%
$$

There is no imbalance in the classes. 